src/i_video.c

```diff
@@ -1423,6 +1423,67 @@ void I_GetScreenDimensions(void)
    WIDESCREENDELTA = (SCREENWIDTH - NONWIDEWIDTH) / 2;
 }
 
+static void CreateUpscaledTexture(int v_w, int v_h)
+{
+    SDL_RendererInfo info;
+    int w, h, w_upscale, h_upscale;
+
+    SDL_GetRendererInfo(renderer, &info);
+
+    if (info.flags & SDL_RENDERER_SOFTWARE)
+    {
+        return;
+    }
+
+    // Get the size of the renderer output. The units this gives us will be
+    // real world pixels, which are not necessarily equivalent to the screen's
+    // window size (because of highdpi).
+
+    SDL_GetRendererOutputSize(renderer, &w, &h);
```

Comment:
Will we need to call this on resize events?

Reply:
Yes, it seem so. To simplify this, maybe we could use [`SDL_GetDesktopDisplayMode`](https://wiki.libsdl.org/SDL2/SDL_GetDesktopDisplayMode) here?

Reply:
I meant on window resize events?

Reply:
> I meant on window resize events?

Yes. I think we can create textures for maximum screen size and let SDL scale them for smaller window sizes.

Reply:
Okay.

Reply:
Done in https://github.com/fabiangreffrath/woof/pull/1019/commits/c4bdc8dfe37ef3207c2b1f7f10bf78bc872797d1